File: RoQui/src/main/kotlin/dev/joguenco/roqui/withhold/model/WithholdDocumentTax.kt
package dev.joguenco.roqui.withhold.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.math.BigDecimal
import java.util.UUID
import org.hibernate.annotations.Immutable

/**
 * Impuestos del documento de sustento (impuestosDocSustento en el XML).
 *
 * No son las retenciones: son los impuestos que ya traia la compra. El XSD V2.0.0 los exige dentro
 * de cada docSustento, agrupados por tarifa.
 */
@Entity
@Immutable
@Table(name = "v_ele_withholds_document_taxes")
class WithholdDocumentTax {

    @Id val id: UUID? = null

    @Column(name = "code") val code: String? = null

    @Column(name = "number") val number: String? = null

    @Column(name = "code_support") val codeSupport: String? = null

    @Column(name = "tax_code") val taxCode: String? = null

    @Column(name = "percentage_code") val percentageCode: String? = null

    @Column(name = "tax_base") val taxBase: BigDecimal? = null

    @Column(name = "tax_iva") val taxIva: BigDecimal? = null

    @Column(name = "value") val value: BigDecimal? = null
}

File: RoQuiApi/RoQui/Document/Model/WithholdDocumentTax.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace RoQuiApi.RoQui.Invoice.Model;

[Table("withholds_document_taxes")]
public class WithholdDocumentTax
{
    [Column("id")]
    public int Id { get; set; }

    [Column("tax_code", TypeName = "varchar")]
    public required string TaxCode { get; set; }

    [Column("percentage_code", TypeName = "varchar")]
    public required string PercentageCode { get; set; }

    [Column("tax_base", TypeName = "decimal")]
    public required decimal TaxBase { get; set; }

    [Column("tax_iva", TypeName = "decimal")]
    public required decimal TaxIva { get; set; }

    [Column("value", TypeName = "decimal")]
    public required decimal Value { get; set; }

    [Column("withhold_support_id")]
    public int WithholdSupportId { get; set; }

    public required WithholdSupport WithholdSupport { get; set; }
}
